llvm/lib/Target/RISCV/RISCVFrameLowering.cpp

```diff
@@ -545,6 +545,16 @@ void RISCVFrameLowering::allocateAndProbeStackForRVV(
       .addReg(SPReg)
       .addReg(TargetReg)
       .setMIFlag(Flag);
+
+  // If we have a dynamic allocation later we need to probe any residuals.
+  MachineBasicBlock *NextMBB = MBBI->getParent()->getSingleSuccessor();
+  if (NextMBB && NextMBB->begin()->getFlag(MachineInstr::FrameSetup)) {
```

Comment:
If understand correctly, this is trying to find the expansion of the `RISCV::PROBED_STACKALLOC_DYN`? What guarantees that would be the first BB after the prologue. Couldn't there be other control flow before it?

Reply:
It's indeed looking for `RISCV::PROBED_STACKALLOC_DYN`.
You're right it could be other BB before it, I'll need to change this and do a longer search.